ttnn/cpp/ttnn/operations/ccl/all_gather/device/multi_core/all_gather_op_multi_core.cpp

```diff
@@ -302,6 +302,7 @@ operation::ProgramWithCallbacks all_gather_multi_core_with_workers_helper(
         worker_defines["INTERLEAVED_MEM_LAYOUT"] = "1";
     }
 
+    constexpr ttnn::ccl::EriscDataMoverTerminationMode edm_termination_mode = ttnn::ccl::EriscDataMoverTerminationMode::WORKER_INITIATED;
```

Comment:
Does all gather not use `EriscDataMoverTerminationMode::MESSAGE_COUNT_REACHED` mode at all now?

Reply:
Correct. Both to simplify host code and to enable migration to fabric we won't be able to use a message_count type of mode anymore.